flasher.cpp

```diff
@@ -255,8 +266,50 @@ void Flasher::LoopHandler()
             break;
         }
 
+        case FlasherStates::kLoadFirmwareFile:
+        {
+            // Load local firmware file
+            if (firmware_file_.size() != 0) {
+                file_content_ = firmware_file_.readAll();
+                firmware_file_.close();
+                firmware_file_.remove();
+                SetState(FlasherStates::kFlash);
+
+                // Load firmware file from url
+            } else {
+
+                DownloadFirmwareFromUrl();
+                emit ShowStatusMsg("Downloading");
+                timer_.start();
+                SetState(FlasherStates::kDownloadFirmwareFile);
+            }
+
+            break;
+        }
+
+        case FlasherStates::kDownloadFirmwareFile:
+        {
+            if (is_firmware_downloaded_) {
+                is_firmware_downloaded_ = false;
+                if (file_content_.isEmpty()) {
+                    emit ShowStatusMsg("Download error");
+                    SetState(FlasherStates::kIdle);
```

Comment:
kError?

Reply:
We should be careful to not degrade functionality. For example, if the download doesn't work Flasher is still capable of browsing the firmware and flashing it.